### ISSUE TITLE
False `almost_complete_letter_range` when the endpoint is matched separately

### ISSUE THREAD

**Issue by cuviper** [rust-lang/rust-clippy#13171]
### Summary

The lint doesn't seem to notice when the endpoint is covered by a separate branch. It doesn't make a difference if I match that endpoint before or after the range pattern.

### Lint Name

almost_complete_letter_range

### Reproducer

I tried this code:

```rust
pub fn rot1(c: char) -> char {
    match c {
        'z' => 'a', // before
        'a'..'z' | 'A'..'Z' => char::from_u32(c as u32 + 1).unwrap(),
        'Z' => 'A', // after
        _ => c,
    }
}
```

I saw this happen:

```
warning: almost complete ascii range
 --> src/lib.rs:4:9
  |
4 |         'a'..'z' | 'A'..'Z' => char::from_u32(c as u32 + 1).unwrap(),
  |         ^^^--^^^
  |            |
  |            help: use an inclusive range: `..=`
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#almost_complete_range
  = note: `#[warn(clippy::almost_complete_range)]` on by default

warning: almost complete ascii range
 --> src/lib.rs:4:20
  |
4 |         'a'..'z' | 'A'..'Z' => char::from_u32(c as u32 + 1).unwrap(),
  |                    ^^^--^^^
  |                       |
  |                       help: use an inclusive range: `..=`
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#almost_complete_range
```

I expected to see this happen: no warning

### Version

```text
rustc 1.82.0-nightly (7120fdac7 2024-07-25)
binary: rustc
commit-hash: 7120fdac7a6e55a5e4b606256042890b36067052
commit-date: 2024-07-25
host: x86_64-unknown-linux-gnu
release: 1.82.0-nightly
LLVM version: 18.1.7
```

**Comment by cuviper** [comment]
For extra context, I had this kind of `rot1` code in my solution for <https://adventofcode.com/2015/day/11>. I previously used an inclusive range pattern like `'a'..='y'`, but I was just trying it with an exclusive range `'a'..'z'` with Rust 1.80.
